e2e/builder/build_test.go

```diff
@@ -57,76 +58,86 @@ func TestKitMaxBuildLimit(t *testing.T) {
-
-		// verify that buildC is allowed to build as soon as buildA has finished
-		Eventually(BuildPhase(ns, buildA), TestTimeoutLong).Should(Equal(v1.BuildPhaseSucceeded))
-		Eventually(BuildPhase(ns2, buildC), TestTimeoutShort).Should(Equal(v1.BuildPhaseRunning))
-		Eventually(KitPhase(ns2, buildC), TestTimeoutLong).Should(Equal(v1.IntegrationKitPhaseBuildRunning))
-
-		// verify that all builds are successful
-		Eventually(BuildPhase(ns1, buildB), TestTimeoutLong).Should(Equal(v1.BuildPhaseSucceeded))
-		Eventually(KitPhase(ns1, buildB), TestTimeoutLong).Should(Equal(v1.IntegrationKitPhaseReady))
-		Eventually(BuildPhase(ns2, buildC), TestTimeoutLong).Should(Equal(v1.BuildPhaseSucceeded))
-		Eventually(KitPhase(ns2, buildC), TestTimeoutLong).Should(Equal(v1.IntegrationKitPhaseReady))
+		WithNewTestNamespace(t, func(ns1 string) {
+			WithNewTestNamespace(t, func(ns2 string) {
+				pl1 := v1.NewIntegrationPlatform(ns1, fmt.Sprintf("camel-k-%s", ns))
+				pl.Spec.DeepCopyInto(&pl1.Spec)
+				pl1.Spec.Build.Maven.Settings = v1.ValueSource{}
+				pl1.SetOperatorID(fmt.Sprintf("camel-k-%s", ns))
+				if err := TestClient().Create(TestContext, &pl1); err != nil {
+					t.Error(err)
+					t.FailNow()
+				}
+
+				Eventually(PlatformPhase(ns1), TestTimeoutMedium).Should(Equal(v1.IntegrationPlatformPhaseReady))
+
+				pl2 := v1.NewIntegrationPlatform(ns2, fmt.Sprintf("camel-k-%s", ns))
+				pl.Spec.DeepCopyInto(&pl2.Spec)
+				pl2.Spec.Build.Maven.Settings = v1.ValueSource{}
+				pl2.SetOperatorID(fmt.Sprintf("camel-k-%s", ns))
+				if err := TestClient().Create(TestContext, &pl2); err != nil {
+					t.Error(err)
+					t.FailNow()
+				}
+
+				Eventually(PlatformPhase(ns2), TestTimeoutMedium).Should(Equal(v1.IntegrationPlatformPhaseReady))
+
+				doKitBuildInNamespace(buildA, ns, TestTimeoutShort, kitOptions{
+					operatorID: fmt.Sprintf("camel-k-%s", ns),
+					dependencies: []string{
+						"camel:timer", "camel:log",
+					},
+					traits: []string{
+						"builder.properties=build-property=A",
+					},
+				}, v1.BuildPhaseRunning, v1.IntegrationKitPhaseBuildRunning)
+
+				doKitBuildInNamespace(buildB, ns1, TestTimeoutShort, kitOptions{
+					operatorID: fmt.Sprintf("camel-k-%s", ns),
+					dependencies: []string{
+						"camel:timer", "camel:log",
+					},
+					traits: []string{
+						"builder.properties=build-property=B",
+					},
+				}, v1.BuildPhaseRunning, v1.IntegrationKitPhaseBuildRunning)
+
+				doKitBuildInNamespace(buildC, ns2, TestTimeoutShort, kitOptions{
+					operatorID: fmt.Sprintf("camel-k-%s", ns),
+					dependencies: []string{
+						"camel:timer", "camel:log",
+					},
+					traits: []string{
+						"builder.properties=build-property=C",
+					},
+				}, v1.BuildPhaseScheduling, v1.IntegrationKitPhaseNone)
+
+				var notExceedsMaxBuildLimit = func(runningBuilds int) bool {
+					return runningBuilds <= 2
+				}
+
+				limit := 0
+				for limit < 5 && BuildPhase(ns, buildA)() == v1.BuildPhaseRunning {
```

Comment:
I think you need to check all BuildPhases in order to constantly check while any build is still running. I think you can indefinetely loop while any build is running and verify that the number is always lower than the limit. I think this could be a better approach then the `Consistently` usage which seem more appropriate when you want to test something in a period of time (which in our case we do not have).

Reply:
Looping indefinitely on all build phases is dangerous as we have already had the situation where builds keep stuck in Running phase forever. This would mean that the test never ends and we have a hanging build. So at some point you always need to break the loop. 

I'd prefer to have a defined end of the test and to make sure that the max build limit checks has been constantly verified as long as the 1st scheduled build is in running phase.

Reply:
For the sake of test this is not a problem as, in case it hangs indefinitely, there is a timeout on the whole suite of tests.

Reply:
I doubt that error messages and failure analysis is clear when running into the test suite timeout. I'd avoid to use this option